### DISCUSSION TITLE
Change Notification Library from react-toastify to Notistack

### DISCUSSION THREAD

**Discussion by moizsheraz** [Infisical/infisical#2538] [General]
### Feature Description
Change the notification library in Infisical from `react-toastify` to `Notistack` in the `frontend/src/components/notifications/Notifications.tsx` file.

### Why Would It Be Useful?
Notistack provides a more modern and visually appealing notification experience, with better customization and management options.

### Additional Context
`react-toastify` feels ordinary, and switching to Notistack could enhance user engagement . More info on Notistack can be found [here](https://notistack.com/).

**Comment by moizsheraz** [comment]
@maidul98 I would like to take on the implementation of this feature.

**Comment by akhilmhdh** [comment]
@moizsheraz react-toastify has been working great for us. 

I don't understand what you mean by better notification experience. Something that has been working quite fine why would we change it? Also react-toastify is much mature than notistack both in terms downloads and usage. We are keeping old version of it due to nextjs issue. 

Moving to discussion for time being as this more of a discussion rather than issue

**Reply by moizsheraz** [reply]
@akhilmhdh okay
